core/db.py

```
@@ -106,8 +106,9 @@ def execute(self, statement: Any, *args: Any, **kwargs: Any):
             result = conn.execute(stmt, *args, **kwargs)
             try:
                 conn.commit()
-            except Exception:
+            except Exception as e:
                 # Not all statements require/allow commit; ignore commit errors
+                logger.debug(f"Commit failed (expected for some statements): {e}")
```

Flake8 (7.3.0)

[error] undefined name 'logger'

(F821)